tf-psa-crypto/scripts/tf_psa_crypto_config.py

```diff
@@ -0,0 +1,184 @@
+
+    PSA_WANT_xxx settings are also boolean, but when they are enabled,
+    they expand to a nonzero value. We leave them undefined when they
+    are disabled. (Setting them to 0 currently means to enable them, but
+    this might change to mean disabling them. Currently we just never set
+    them to 0.)
+    """
+    if name.startswith('PSA_WANT_'):
+        return True
+    if not value:
+        return True
+    return False
+
+def include_in_full(name):
+    """Rules for symbols in the "full" configuration."""
+    if name in EXCLUDE_FROM_FULL:
+        return False
+    return True
+
+def full_adapter(name, value, active):
+    """Config adapter for "full"."""
+    if not is_boolean_setting(name, value):
+        return active
+    return include_in_full(name)
+
+
+class TfPSACryptoConfigFile(config_common.ConfigFile):
+    """Representation of a TF PSA Crypto configuration file."""
+
+    _path_in_tree = 'tf-psa-crypto/include/psa/crypto_config.h'
+    default_path = [_path_in_tree,
+                    os.path.join(os.path.dirname(__file__),
+                                 os.pardir,
+                                 _path_in_tree),
+                    os.path.join(os.path.dirname(os.path.abspath(os.path.dirname(__file__))),
+                                 _path_in_tree)]
+
+    def __init__(self, filename=None):
+        super().__init__(self.default_path, 'Crypto', filename)
+
+
+class TfPSACryptoConfig(config_common.Config):
+    """Representation of the TF PSA Crypto configuration.
+
+    See the documentation of the `Config` class for methods to query
+    and modify the configuration.
+    """
+
+    def __init__(self, *configfiles):
+        """Read the PSA crypto configuration files."""
+
+        super().__init__()
+        self.configfiles.extend(configfiles)
+        self.settings.update({name: config_common.Setting(configfile, active, name, value, section)
+                             for configfile in configfiles
+                             for (active, name, value, section) in configfile.parse_file()})
+
+    def set(self, name, value=None):
+        """Set name to the given value and make it active."""
+
+        if name in PSA_UNSUPPORTED_FEATURE:
+            raise ValueError(f'Feature is unsupported: \'{name}\'')
+        if name in PSA_UNSTABLE_FEATURE:
+            raise ValueError(f'Feature is unstable: \'{name}\'')
+
+        if name not in self.settings:
+            self._get_configfile().templates.append((name, '', f'#define {name} '))
+
+        # Default value for PSA macros is '1'
+        if name.startswith('PSA_') and not value:
+            value = '1'
+
+        super().set(name, value)
+
+
+class TfPSACryptoConfigTool(config_common.ConfigTool):
+    """Command line TF PSA Crypto config file manipulation tool."""
+
+    def __init__(self):
+        super().__init__(TfPSACryptoConfigFile.default_path[0], single_config=False)
```

Comment:
There is only one configuration file in TF-PSA-Crypto, namely `include/psa/crypto_config.h`, thus we do not need support for multiple configuration files. Please remove that support. We do not need it and things are already complex enough without.